### DISCUSSION TITLE
The state store, topic-store, may have migrated to another instance

### DISCUSSION THREAD

**Discussion by Muddam123** [strimzi/strimzi-kafka-operator#8527] [Q&A]
Hi All,

I have existing topic and i have added the retention period for 2 days and then deployed kafka topic then i saw the status is not ready in kubernetes cluster. Can any one please help on this issue and this issue i am facing production. when i describe the topic i saw this message.

before my YML:
apiVersion: kafka.strimzi.io/v1beta2
kind: KafkaTopic
metadata:
  name: "prod-XXXX-order-XXXXX-XXXX"
  labels:
    strimzi.io/cluster: prod-XXXXXXXXXXXXXXX
spec:
  partitions: 6
  replicas: 3

Added retention period YML:
apiVersion: kafka.strimzi.io/v1beta2
kind: KafkaTopic
metadata:
  name: "prod-XXXX-order-XXXXX-XXXX"
  labels:
    strimzi.io/cluster: prod-XXXXXXXXXXXXXXX
spec:
  partitions: 6
  replicas: 3
  config:
      retention.ms: 7200000
      segment.bytes: 1073741824

when i describe:
  Conditions:
    Last Transition Time:  2023-05-18T08:51:06.214844Z
    Message:               The state store, topic-store, may have migrated to another instance.
    Reason:                InvalidStateStoreException
    Status:                True
    Type:                  NotReady
  Observed Generation:     1
  Topic Name:              prod-order-enriched
Events:
  Type     Reason  Age        From                                     Message
  ----     ------  ----       ----                                     -------
  Warning          <unknown>  io.strimzi.operator.topic.TopicOperator  Failure processing KafkaTopic watch event ADDED on resource prod-cn-order-item-enriched with labels {strimzi.io/cluster=prod-oc4-selling-kafka-alicloud}: The state store, topic-store, may have migrated to another instance.

**Comment by Muddam123** [comment]
@scholzj Can you please help me on this issue

**Reply by scholzj** [reply]
I have no idea about Topic Operator I'm afraid.
